Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp

```diff
@@ -101,84 +101,74 @@ Ref<MediaPromise> RemoteSourceBufferProxy::sourceBufferPrivateDidReceiveInitiali
 
     segmentInfo.audioTracks = segment.audioTracks.map([&](auto& audioTrackInfo) {
         auto id = audioTrackInfo.track->id();
-        m_remoteMediaPlayerProxy->addRemoteAudioTrackProxy(*audioTrackInfo.track);
+        m_remoteMediaPlayerProxy->addRemoteAudioTrackProxy(Ref { *audioTrackInfo.track });
```

Comment:
We should be adding `audioTrackInfo.protectedTrack()` function instead.

Reply:
`audioTrackInfo.track` already returns a RefPtr. Should I add a `protectedTrack` function like so:
`struct AudioTrackInformation {
          RefPtr<MediaDescription> description;
          RefPtr<AudioTrackPrivate> track;

          Ref<AudioTrackPrivate> protectedTrack() const { return *track; }
      };`

Reply:
oh, no. Adding a version which returns RefPtr is good so like this:
```
struct AudioTrackInformation {
     RefPtr<MediaDescription> description;
     RefPtr<AudioTrackPrivate> track;

     RefPtr<AudioTrackPrivate> protectedTrack() const { return track; }
};
```

Calling `protectedTrack()` is different from accessing `track` in that it creates a temporary RefPtr which ref's `AudioTrackPrivate`.

Reply:
Ohh ok got it. That makes more sense.